README.md

```diff
@@ -1,358 +1,302 @@
+taskEither.fromEither(eitherValue); // => taskEither.right(42)
 
-```typescript
-import { readonlyArray } from "fp-ts";
-const array = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+// Build from an async function
+const asyncIsEven = async (n: number) => {
+  await sleep(1000);
 
-pipe(array, readonlyArray.filter(isEven), readonlyArray.map(square));
-// => [0, 4, 16, 36, 64]
-```
+  if (!isEven(n)) {
+    throw new Error(`${n} is odd`);
+  }
 
-#### <a name="filter-map"></a>Filter and map in one go
+  return n;
+};
 
-Sometimes, you want to map an array with a function that may fail to produce a value and then get rid of those failures. That is precisely what `array.filterMap` does:
+const buildTaskEither = (n: number) =>
+  taskEither.tryCatch(
+    () => asyncIsEven(n),
+    (error: Error) => error.message
+  );
 
-```typescript
-// getById(id: Id): Option<Item>;
-const ids = [id1, id2, id3];
-const foundItems = pipe(ids, array.filterMap(getById));
-// => Array<Item>
+buildTaskEither(3); // => taskEither.left('3 is odd')
+buildTaskEither(4); // => taskEither.right(4)
 ```
 
-#### <a name="sort"></a>Sorting with `Ord` instances
+#### <a name="taskeither-getters"></a>Extract inner value
 
-Sorting strings:
+```ts
+// Simple case: an infaillible Task
+const someTask = task.of(42);
 
-```typescript
-import { array, ord, string } from "fp-ts";
-
-const strings = ["zyx", "abc", "klm"];
+// Invoking a Task returns the underlying Promise
+await someTask(); // => 42
 
-const sortedStrings = pipe(strings, array.sort(string.Ord));
-// => ['abc', 'klm', 'zyx']
+// TaskEither
+const someTaskEither = taskEither.right(42);
+const eitherValue = await someTaskEither(); // => either.right(42)
+either.getOrElse(() => 0)(eitherValue); // => 42
 ```
 
-There are various instances of `Ord` for primitive types, available in `fp-ts`:
-
-- Strings: `string.Ord: Ord<string>`
-- Numbers: `number.Ord: Ord<number>`
-- Booleans: `boolean.Ord: Ord<boolean>`
-- Dates: `date.Ord: Ord<Date>`
+## <a name="array"></a>Array and ReadonlyArray
 
-The `Ord` module also provides us with ways to manipulate and combine them to create richer sorting algorithms!
+### <a name="basic-manipulation"></a>Basic manipulation
 
-For example, to get an instance of `Ord` that will sort strings in reverse order, I can simply create and use it as such:
-
-```typescript
-const strings = ["zyx", "abc", "klm"];
+```ts
+const someArray = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
 
-const reversedOrdString = ord.reverse(string.Ord);
+// filter and map
+const result = pipe(
+  someArray,
+  array.filter(isEven),
+  array.map(square)
+); // => [0, 4, 16, 36, 64]
 
-const sortedStrings = pipe(strings, array.sort(reversedOrdString));
-// => ['zyx', 'klm', 'abc']
+// or in one go with filterMap
+const result = pipe(
+  someArray,
+  array.filterMap(
+    flow(option.fromPredicate(isEven), option.map(square))
+  )
+);
```

Comment:
I think it would have some value to add example for `findFirst`, `compact`, `head` and `tail`